pkg/querier/blocks_store_queryable.go

```diff
@@ -611,6 +827,69 @@ func createSeriesRequest(minT, maxT int64, matchers []storepb.LabelMatcher, skip
 	}, nil
 }
 
+func createLabelNamesRequest(minT, maxT int64, blockIDs []ulid.ULID) (*storepb.LabelNamesRequest, error) {
+	req := &storepb.LabelNamesRequest{
+		Start: minT,
+		End:   maxT,
+	}
+
+	if len(blockIDs) == 0 {
```

Comment:
Is this a real case? When does it happen?

Reply:
When Thanos querier queries our storegateways. I know it's a rare-case (maybe even non-existent) and can remove it if you'd like.